tests/test_type_coverage_generation.py

```diff
@@ -0,0 +1,214 @@
+from __future__ import annotations
+
+import sys
+from dataclasses import dataclass
+from datetime import date
+from typing import Literal, Sequence
+from uuid import UUID
+
+import pytest
+from typing_extensions import TypedDict
+
+from polyfactory.decorators import post_generated
+from polyfactory.factories.dataclass_factory import DataclassFactory
+from polyfactory.factories.typed_dict_factory import TypedDictFactory
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="requires python3.10 or higher")
```

Comment:
Why do these tests require 3.10+?

Reply:
https://github.com/litestar-org/polyfactory/actions/runs/6529149394/job/17726381942?pr=390

I'm defining some union types in the tests using the 3.10+ syntax `A | B`, attempting to use the older syntax `Union[A, B]` does not work because the linters seem to automatically rewrite to the new syntax